### PR TITLE
#11962: confirmMessage facet

### LINKED ISSUE #11962: Confirm: message facet
### Description

The `p:confirm` `message` attribute is quite limited to express formatted content.
Hence we add support for the following construct:
```xml
<p:commandButton ...>
    <p:confirm header="title"/>
    <f:facet name="confirmMessage">
        fancy content here...
    </f:facet>
</p:commandButton>
```

### Describe the solution you would like

`p:confirm` uses the `confirmMessage` facet of its parent to yield the message content.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by fcorneli**
Fix #11962

### CODE REVIEW

File: primefaces/src/main/java/org/primefaces/behavior/confirm/ConfirmBehavior.java [hunk 1]
```diff
@@ -78,7 +81,26 @@ public String getScript(ClientBehaviorContext behaviorContext) {
         String source = getSource();
         String type = JSONObject.quote(getType());
         String headerText = JSONObject.quote(getHeader());
-        String messageText = JSONObject.quote(getMessage());
+
+        String messageText;
+        UIComponent messageFacetComponent = component.getFacet("confirmMessage");
+        if (null != messageFacetComponent) {
```
Comment (melloware) [inline primefaces/src/main/java/org/primefaces/behavior/confirm/ConfirmBehavior.java]:
```suggestion
        if (FacetUtils.shouldRenderFacet(messageFacetComponent)) {
```

File: primefaces/src/main/java/org/primefaces/behavior/confirm/ConfirmBehavior.java [hunk 2]
```diff
@@ -78,7 +81,26 @@ public String getScript(ClientBehaviorContext behaviorContext) {
...
+            ResponseWriter originalWriter = context.getResponseWriter();
+            FastStringWriter fsw = new FastStringWriter();
+            ResponseWriter clonedWriter = originalWriter.cloneWithWriter(fsw);
+            context.setResponseWriter(clonedWriter);
+            try {
+                messageFacetComponent.encodeAll(context);
+            }
+            catch (IOException ex) {
+                fsw.write("error encoding confirm message facet");
```
Comment (melloware) [inline primefaces/src/main/java/org/primefaces/behavior/confirm/ConfirmBehavior.java]: I think we typically just let these kinds of errors just bubble up?

**Review by melloware — CHANGES_REQUESTED** [review]

File: primefaces/src/main/java/org/primefaces/behavior/confirm/ConfirmBehavior.java (hunk 2 above)
Comment (melloware) [reply primefaces/src/main/java/org/primefaces/behavior/confirm/ConfirmBehavior.java]: Actually use `ComponentUtils.encodeComponent` which wraps a lot of this logic up nicely!

Comment (fcorneli) [reply primefaces/src/main/java/org/primefaces/behavior/confirm/ConfirmBehavior.java]: Thanks for the review. Indeed nice utility functions. Will use these.

**Review by melloware — APPROVED** [review]

**Pull request merged**